cmd/arena/commands/diagnose.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kubeflow/arena/pkg/types"
+	"github.com/kubeflow/arena/pkg/util"
+
+	"k8s.io/api/apps/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type DiagnoseArgs struct {
+	outputDir string
+}
+
+var (
+	env = os.Environ()
+)
+
+const(
+	DIR_PATH = "/tmp/"
+	KUBECTL_BIN = "kubectl"
+	KUBE_DESCRIBE = "describe"
+	KUBE_GET = "get"
+	KUBE_LOGS = "logs"
+	KUBE_EVENTS = "events"
+	KUBE_POD = "pod"
+	KUBE_STS = "statefulsets"
+	KUBE_JOB = "jobs"
+	KUBE_NODE = "nodes"
+)
+
+func NewDiagnoseCommand() *cobra.Command {
+	diagnoseArgs := DiagnoseArgs{}
+
+	var command = &cobra.Command{
+		Use:   "diagnose job",
+		Short: "diagnose relevant logs of a job",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
+			jobName := args[0]
+
+			util.SetLogLevel(logLevel)
+			setupKubeconfig()
+			client, err := initKubeClient()
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			err = updateNamespace(cmd)
+			if err != nil {
+				log.Errorf("Failed due to %v", err)
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			path := DIR_PATH
+			if len(diagnoseArgs.outputDir) != 0 {
+				path = diagnoseArgs.outputDir
```

Comment:
I think it's better to create a sub directory with the name like jobname + jobtype +timestamp, because we need distinguish the different result. Thanks.